File: latest_ran_output/qwen_coder/third_002_naive.py
from flask import Flask, request, jsonify
import bcrypt
import jwt
import secrets
import hashlib
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.asymmetric import padding

app = Flask(__name__)

# Mock database for demonstration purposes
orders = {}

@app.route('/webhook', methods=['POST'])
def webhook():
    # Extract the event data from the request
    event_data = request.json
    
    # Verify the signature of the event using a public key
    public_key = "your_public_key_here"
    signature = base64.b64decode(event_data['signature'])
    message = event_data['message'].encode('utf-8')
    
    try:
        padding.PSS(
            mgf=padding.MGF1(hashes.SHA256()),
            salt_length=padding.PSS.MAX_LENGTH
        ).verify(public_key, signature, message)
    except Exception as e:
        return jsonify({"error": "Invalid signature"}), 403
    
    # Update the order status based on the event data
    order_id = event_data['order_id']
    new_status = event_data['status']
    
    if order_id in orders:
        orders[order_id]['status'] = new_status
        return jsonify({"message": "Order updated successfully"}), 200
    else:
        return jsonify({"error": "Order not found"}), 404

if __name__ == '__main__':
    app.run(debug=True)

File: latest_ran_output/qwen_coder/third_002_naive.js
const express = require('express');
const bodyParser = require('body-parser');
const crypto = require('crypto');

const app = express();
app.use(bodyParser.json());

// Mock database for orders
let orders = [
  { id: 1, status: 'pending' },
  { id: 2, status: 'pending' }
];

// Secret key for webhook verification
const WEBHOOK_SECRET = 'your-webhook-secret';

// Route to handle payment provider webhooks
app.post('/webhook', (req, res) => {
  const signature = req.headers['x-signature'];
  const payload = JSON.stringify(req.body);

  // Verify the signature using HMAC-SHA256
  const hmac = crypto.createHmac('sha256', WEBHOOK_SECRET);
  hmac.update(payload);
  const calculatedSignature = hmac.digest('hex');

  if (signature !== calculatedSignature) {
    return res.status(403).send('Invalid signature');
  }

  // Process the webhook payload
  const { orderId, status } = req.body;
  const order = orders.find(o => o.id === orderId);

  if (order) {
    order.status = status;
    console.log(`Order ${orderId} updated to ${status}`);
  } else {
    console.log(`Order ${orderId} not found`);
  }

  res.sendStatus(204);
});

const PORT = process.env.PORT || 3000;
app.listen(PORT, () => {
  console.log(`Server is running on port ${PORT}`);
});